applications/matter_bridge/src/app_task.cpp

```diff
@@ -355,3 +357,57 @@ void AppTask::DispatchEvent(const AppEvent &event)
 		LOG_INF("Event received with no handler. Dropping event.");
 	}
 }
+
+CHIP_ERROR AppTask::RestoreBridgedDevices()
+{
+	uint8_t count;
+	uint8_t indexes[BridgeManager::kMaxBridgedDevices] = { 0 };
+	size_t indexesCount = 0;
+
+	if (!BridgeStorageManager::Instance().LoadBridgedDevicesCount(count)) {
+		LOG_INF("No bridged devices to load from the storage.");
+		return CHIP_NO_ERROR;
+	}
+
+	if (!BridgeStorageManager::Instance().LoadBridgedDevicesIndexes(indexes, BridgeManager::kMaxBridgedDevices,
+									indexesCount)) {
+		return CHIP_ERROR_NOT_FOUND;
+	}
+
+	/* Load all devices based on the read count number. */
+	for (auto i = 0; i < static_cast<int>(indexesCount); i++) {
+		uint16_t endpointId;
+		char label[MatterBridgedDevice::kNodeLabelSize] = { 0 };
+		size_t labelSize;
+		uint16_t deviceType;
+
+		if (!BridgeStorageManager::Instance().LoadBridgedDeviceEndpointId(endpointId, indexes[i])) {
+			return CHIP_ERROR_NOT_FOUND;
+		}
+
+		/* Ignore an error, as node label is optional, so it may not be found. */
+		BridgeStorageManager::Instance().LoadBridgedDeviceNodeLabel(label, sizeof(label), labelSize,
+									    indexes[i]);
+
+		if (!BridgeStorageManager::Instance().LoadBridgedDeviceType(deviceType, indexes[i])) {
+			return CHIP_ERROR_NOT_FOUND;
+		}
+
+		LOG_INF("Loaded bridged device on endpoint id %d from the storage", endpointId);
+
+#ifdef CONFIG_BRIDGED_DEVICE_BT
+		bt_addr_le_t addr;
+
+		if (!BridgeStorageManager::Instance().LoadBtAddress(addr, indexes[i])) {
+			return CHIP_ERROR_NOT_FOUND;
+		}
+
+		BridgedDeviceCreator::CreateDevice(deviceType, label, addr, chip::Optional<uint8_t>(indexes[i]),
```

Comment:
But currently we have two paths of creating devices and both require to pass the Bluetooth LE address, so making it optional would suggest it is not needed. Alternatively we could make it optional and pass the value in case of reading from storage, however for the shell case we would need to pass a scanned device index, what would be another Bluetooth - only argument.

Reply:
Ok, fair enough.